### PR TITLE
fix: view for assessor methods

### PR DESCRIPTION
**Pull request by manuelpolzhofer**
- senior.debt() returns the latest debt but doesn't modify the state anymore

-  calcUpdatedTokenPrice, redeemApprove, supplyApprove update the state by calling drip before using the view methods

- if we use one of the assessor methods in a transaction we should always update debt before

### CODE REVIEW

File: src/lender/tranche/senior_tranche.sol [hunk 1]
```diff
@@ -36,17 +36,21 @@ contract SeniorTranche is Tranche, Interest {
...
     }
 
+    function updatedDebt() external returns(uint) {
+        drip();
+        return safeAdd(borrowed, interest);
+    }
+
+    function debt() external view returns(uint) {
+        return safeAdd(borrowed, _calcInterest());
+    }
```
Comment (manuelpolzhofer) [inline src/lender/tranche/senior_tranche.sol]: biggest change.

all other contracts should use `updatedDebt` only the assessor uses `debt` as view. (and might call `drip` before)

File: src/lender/assessor/base.sol [hunk 2]
```diff
@@ -185,6 +199,7 @@ contract BaseAssessor is Math, Auth {
         if (tranche == junior || minJuniorRatio == 0) {
             return true;
         }
+        _drip();
```
Comment (manuelpolzhofer) [inline src/lender/assessor/base.sol]: supplyApprove should update the debt as well.

File: src/lender/assessor/base.sol [hunk 3]
```diff
@@ -201,6 +216,8 @@ contract BaseAssessor is Math, Auth {
             return true;
         }
 
+        _drip();
```
Comment (manuelpolzhofer) [inline src/lender/assessor/base.sol]: same for redeem approve

File: src/lender/assessor/base.sol [hunk 4]
```diff
@@ -81,22 +83,34 @@ contract BaseAssessor is Math, Auth {
...
+        return _calcSeniorAssetValue(poolValue, trancheReserve, _seniorDebt(), _juniorReserve());
     }
 
-    function calcTokenPrice(address tranche) public returns (uint) {
+    function calcTokenPrice(address tranche) public view returns (uint) {
         require(tranche  == junior || tranche == senior);
         return safeMul(_calcTokenPrice(tranche), tokenAmountForONE);
     }
+    
+    function calcUpdatedTokenPrice(address tranche) public returns (uint) {
```
Comment (manuelpolzhofer) [inline src/lender/assessor/base.sol]: all other contracts should use this method

File: src/lender/assessor/base.sol (hunk 3 above)
Comment (ilinzweilin) [reply src/lender/assessor/base.sol]: I think it would make sense to make this function view as well

File: src/lender/assessor/base.sol (hunk 2 above)
Comment (ilinzweilin) [reply src/lender/assessor/base.sol]: same here, having this function as a view would be really helpful

Comment (charleenfei) [reply src/lender/assessor/base.sol]: *note

File: src/lender/assessor/base.sol (hunk 3 above)
Comment (charleenfei) [reply src/lender/assessor/base.sol]: *note

**Pull request merged**